### ISSUE TITLE
Transformer as encoder, lstm as decoder

### ISSUE THREAD

**Issue by unbreading** [OpenNMT/OpenNMT-tf#438]
I wanna use OpenNMT to build a transformer encoder and lstm(or gru) decoder model, but I do not know how to config.
Could I simply feed SelfAttentionEncoder and AttentionalRNNDecoder to SequenceToSequence? I've tried many times but always got exception.
How could I config correctly? Or any paper / people already done it?

**Comment by guillaumekln** [comment]
What did you try so far?

**Comment by unbreading** [comment]
Generally, I created a SequenceToSequence instance using SelfAttentionEncoder as encoder and AttentionalRNNDecoder as decoder.
The mainly exeception i got: 
"The two structures(Transformer/LSTM) don't have the same nested structure" in "tensorflow/python/util/nest.py" 
I guess this is because LSTM/GRU-decoder always need a global sentence feature to initialize its h_0/c_0, however Transformer does not explicitly generate a global sentence feature. One solution is adding a "[ctx]" token at the beginning of source sentence then using the "[ctx]"-token's feature as the global sentence feature. I also tried this, but it's really hard for me to implement this in a short time.
